staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -660,6 +697,170 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 	return nil
 }
 
+// Saves the last-applied-configuration annotation in a separate SSA field manager
+// to prevent it from being dropped by users who have transitioned to SSA.
+//
+// If this operation is not performed, then the last-applied-configuration annotation
+// would be removed from the object upon the first SSA usage.
+//
```

Comment:
Is it worth mentioning _why_ we want to keep it even though the user is on SSA? I.e. that it is required for downgrade to be possible (per the [docs](https://kubernetes.io/docs/reference/using-api/server-side-apply/#downgrading-from-server-side-apply-to-client-side-apply) and the [KEP](https://github.com/kubernetes/kubernetes/pull/90187))

Reply:
added doc links